src/LeitnerIntance.ts

```diff
@@ -0,0 +1,89 @@
+import { LeitnerColl } from '@/models/LeitnerColl';
+import { Puzzle } from '@/types/lichess-api';
+import { User } from 'lucia';
+
+const BOX_A_PROB: number = 0.8;
+const BOX_LIMIT: number = 50;
+
+interface LeitnerInstance {
+  boxA: Array<Puzzle>;
+  boxB: Array<Puzzle>;
+}
+
+const findLeitner = async (
+  user: User
+): Promise<LeitnerInstance | undefined> => {
+  const leitner = await LeitnerColl.findOne({ username: user.username });
+  return leitner ? { boxA: leitner.boxA, boxB: leitner.boxB } : undefined;
```

Comment:
It's possible to return empty boxes here, it'd simplify the code, but I decided to mirror Lichess.